internal/cache/avi_ctrl_clients.go

```diff
@@ -34,8 +34,9 @@ func SharedAVIClients() *utils.AviRestClientPool {
 
 	ctrlUsername := os.Getenv("CTRL_USERNAME")
 	ctrlPassword := os.Getenv("CTRL_PASSWORD")
+	ctrlAuthtoken := os.Getenv("CTRL_AUTHTOKEN")
 	ctrlIpAddress := os.Getenv("CTRL_IPADDRESS")
-	if ctrlUsername == "" || ctrlPassword == "" || ctrlIpAddress == "" {
+	if ctrlUsername == "" || (ctrlPassword == "" && ctrlAuthtoken == "") || ctrlIpAddress == "" {
```

Comment:
Do we need the username when auth token is used ?

Reply:
authtoken is associated with the username, and used same as password